include/range/v3/experimental/view/shared.hpp

```diff
@@ -0,0 +1,133 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Filip Matzner 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+#ifndef RANGES_V3_VIEW_SHARED_HPP
+#define RANGES_V3_VIEW_SHARED_HPP
+
+#include <memory>
+#include <type_traits>
+#include <meta/meta.hpp>
+#include <range/v3/begin_end.hpp>
+#include <range/v3/iterator_range.hpp>
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/size.hpp>
+#include <range/v3/view/all.hpp>
+
+namespace ranges
+{
+namespace experimental
+{
+    inline namespace v3
+    {
```

Comment:
The namespace should be `ranges::v3::experimental`, not `ranges::experimental::v3`. See ranges/v3/experimental/utility/generator.hpp (which should probably also be moved into the experimental/view/ directory now that it's a view).